src/osp/Resource/PackageRegistry.h

```diff
@@ -24,49 +24,58 @@
  */
 #pragma once
 
-#include <map>
-#include <spdlog/spdlog.h>
+#include "Package.h"
 
-#include "Universe.h"
-#include "Resource/Package.h"
+#include <map>
 
 namespace osp
 {
 
+// Prefix for resource paths to indicate which package they're from
+// Ideally, this would be some kind of short string
+using ResPrefix_t = std::string;
```

Comment:
boost::static_string, or std::array.

Reply:
The prefixes are typically short, so small string optimization will probably be able to contain the whole thing in 99% of cases. `std::basic_string` has a buffer size of 16 chars in the STL implementation I've got for MSVC, and I think the gcc equivalent is even longer. The prefixes were originally going to be only 4 characters but 16 is a ton so I think it's probably fine. If we get to the point where `std::string` would really be a bad idea here, we'd be moving away from string-based keys anyways.

Reply:
std::string here works for now; it's not in a performance critical path either. Regardless of whatever implementation is used, stuff is still going to run a couple hundred times faster than a python program. I'd leave this as is, but open to a drive-by PR.